client/state/breadcrumb/reducer.js

```diff
@@ -24,3 +25,5 @@ export default ( state = [], action ) => {
 
 	return state;
 };
+
+export default keyedReducer( 'siteId', breadcrumbs );
```

Comment:
Yes, the experience should be site dependant like it is for the sidebar. If that breadcrumb was more generic showing the navigation a user does inside Calypso, then we would want to keep all tokens. Now though, breadcrumb is related to the parent page which in this case is Plugins. If a user has selected to switch site, they would expect that all links in that page refer to that site. Do you agree?

Reply:
>  If a user has selected to switch site, they would expect that all links in that page refer to that site. Do you agree?

Yes, I would expect all links to point to the new site, but now I'm losing the previous references if I switch sites. Eg: "Search Results"
![Kapture 2022-03-11 at 12 31 39](https://user-images.githubusercontent.com/5039531/157898002-1c793544-50ef-4d9f-bce4-957e60bdfec2.gif)

Reply:
Yeah, but that previous reference pointed to the previous site and not to the current selected one.

Reply:
There is a way to keep the breadcrumb but pointing to the new site?

Reply:
From the UX standpoint, I think may not be clear to the user the breadcrumbs are site dependant, and that it will ended up in a previous state of the selected site. Eg: 
* Search for something(lets say **A**) with one site selected (or none)
* Go to the plugin and switch site
* Go back to plugins and search a new plugin (lets say **B**)
* Go back to the previous site
* Clicking on Search results will lead you to search **A**


![Kapture 2022-03-14 at 12 04 34](https://user-images.githubusercontent.com/5039531/158201248-17fe1b0e-762b-4d59-bb43-14499015c35e.gif)

Reply:
Cool! Thanks for pushing on broader discussion here! Posted here pdt2If-5P-p2